### ISSUE TITLE
fit_transform hangs for low number of samples

### ISSUE THREAD

**Issue by fredmell** [YingfanWang/PaCMAP#80]
Hello,

Thank you for this great package! It seems like I have run into a bug with the most recent version.

`fit_transform` hangs indefinitely for `n=30` and less. It runs fine for values larger than this.

```
import numpy as np
import pacmap 

n = 20
X = np.random.normal(size=(n, 512))

embedding = pacmap.PaCMAP(n_components=2) 

X_transformed = embedding.fit_transform(X)

print(X_transformed.shape)
```

It seems to have been introduced in `0.7.4`, it runs to completion on `0.7.3`.

Let me know if there is anything more I can provide.

**Comment by hadronzoo** [comment]
This might be caused by the `sample_FP` function entering an infinite loop if `n_samples > maximum - len(reject_ind)`.

**Comment by hyhuang00** [comment]
Resolved in release 0.7.6. Please reopen this issue if you encountered this issue again.
